docs/madr/decisions/017-mesh-proxy-template.md

```diff
@@ -0,0 +1,244 @@
+* MeshEnvoyConfig
+* MeshXDSConfig
+* MeshEnvoyMods (but the policy can potentially include other stuff than modifications in the future)
+* MeshProxyConfig (too generic? sounds like configuration of Kuma DP)
+* MeshProxyPatch
+
+While `MeshProxyPatch` seems to be the most accurate name, I'm hesitant to make this change. I think that changing the name may confuse existing users. 
+
+#### Other resources
+
+##### Transport Socket modifications
+
+We could introduce "native" transport_socket modification, but if we introduce `jsonPatches` we can also rely on it.
+
+```yaml
+appendModifications:
+  - transport_socket:
+      operation: add # (or patch or remove)
+      match:
+        name: "envoy.transport_sockets.tls" # optional: name of the transport socket match on which to apply modification (can be used only with patch and remove) 
+        origin: outbound # optional: origin of the resource on which to apply the modification (ex. inbound, outbound)
+        resource: cluster # optional: resource type on which to apply the modification (available values: listener, cluster)
+        resourceName: "xyz" # optional: exact name of the resource on which to apply the modification
+      value: |
+        name: envoy.transport_sockets.tls
+        typedConfig:
+          '@type': 'type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext'
+          commonTlsContext:
+            tlsParams:
+              tlsMinimumProtocolVersion: TLSv1_2
+            validationContextSdsSecretConfig:
+              name: cp_validation_ctx
+          sni: kuma-control-plane.kuma-system
+```
+
+Example of overriding TLS version for mesh mTLS
+
+```yaml
+appendModifications:
+  - transport_socket:
+      operation: patch
+      match:
+        name: "envoy.transport_sockets.tls" 
+        origin: outbound
+        resource: cluster
+      value: |
+        name: envoy.transport_sockets.tls
+        typedConfig:
+          '@type': 'type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext'
+          commonTlsContext:
+            tlsParams:
+              tlsMinimumProtocolVersion: TLSv1_3
+  - transport_socket:
+      operation: patch
+      match:
+        name: "envoy.transport_sockets.tls"
+        origin: inbound
+        resource: listener
+      value: |
+        name: envoy.transport_sockets.tls
+        typedConfig:
+          '@type': 'type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.DownstreamTlsContext'
+          commonTlsContext:
+            tlsParams:
+              tlsMinimumProtocolVersion: TLSv1_3
+```
+
+##### Route modifications
+
+Route modifications will support ordered modifications by allowing the same operations as `httpFilter` - `addFirst`, `addBefore`, `addAfter`, `addLast`, `patch`, `remove`.
+Route modifications will support static routes (in-cluster communication) and dynamic routes through RDS (gateway).
+To support name matching, we also need to name all the routes from now.
```

Comment:
This is one for @michaelbeaumont

Reply:
is this something we expose to the user to potentially change or is it just a name that can be derived deterministically from the route resources?

Reply:
some of the routes do not have names now. We need to change this in whatever way we want.